File: app/src/main/java/mk/webfactory/template/data/storage/StorageException.java
package mk.webfactory.template.data.storage;

public class StorageException extends RuntimeException {

  public StorageException() {
  }

  public StorageException(String detailMessage) {
    super(detailMessage);
  }

  public StorageException(String detailMessage, Throwable throwable) {
    super(detailMessage, throwable);
  }

  public StorageException(Throwable throwable) {
    super(throwable);
  }
}

File: app/src/main/java/mk/webfactory/template/data/storage/StorageException.kt
package mk.webfactory.template.data.storage

class StorageException : RuntimeException {
    constructor()
    constructor(detailMessage: String?) : super(detailMessage)
    constructor(detailMessage: String?, throwable: Throwable?) : super(detailMessage, throwable)
    constructor(throwable: Throwable?) : super(throwable)
}
